tests/Model/PageTest.php

```diff
@@ -28,7 +28,11 @@ class PageTest extends TestCase
     public function testSlugify(): void
     {
         setlocale(LC_ALL, 'en_US.utf8');
-        setlocale(LC_CTYPE, 'en_US.utf8');
+
+        $reflectionClass = new \ReflectionClass(Page::class);
+        $property = $reflectionClass->getProperty('slugifyMethod');
+        $property->setAccessible(true);
+        $property->setValue(null);
```

Comment:
This test was failing because with the functional test, when this bundle is loaded, [`Page::slugifyMethod` is set](https://github.com/sonata-project/SonataPageBundle/blob/3.x/src/SonataPageBundle.php#L64). So it was using the cocur slugify method provided by the CoreBundle.

Reply:
If we need to check the implementation provided by this bundle, should not be better to use the corresponding configuration?
https://github.com/sonata-project/SonataPageBundle/blob/926cbbf124e214cc9b9824051b7273836635fd35/tests/App/config/sonata.yaml#L14

Reply:
[The current implementation is deprecated](https://github.com/sonata-project/SonataPageBundle/blob/3.x/src/Model/Page.php#L590), there is no `slugify_service` in this bundle. [If you don't set it, it will use `sonata.core.slugify.native`](https://github.com/sonata-project/SonataPageBundle/blob/3.x/src/DependencyInjection/Configuration.php#L118). Not sure if this is what you meant 🤔

Reply:
I'm not a user of this bundle, so I haven't experience with its behavior.
Re-readig, I thought this test is functional, but it isn't, right? If not, why is it being affected by a functional test?
>This test was failing because with the functional test, when this bundle is loaded, Page::slugifyMethod is set.

Based in your comment:
>So it was using the cocur slugify method provided by the CoreBundle instead of the own implementation.

I understood what you're confirming here:
>If you don't set it, it will use sonata.core.slugify.native.

About my comment
>If we need to check the implementation provided by this bundle, should not be better to use the corresponding configuration?

I understand there is no service defining a slugger in this bundle. I was referencing the implementation you've described before. What I'm asking is if there is a way to avoid the "hack" to manipulate a property value with via reflection.

Finally, since this test isn't checking all the available implementations, I think it should be renamed in order to make clear which one is intended to checked (own, "sonata.core.slugify.native", or "sonata.core.slugify.cocur").

Reply:
> Re-readig, I thought this test is functional, but it isn't, right? If not, why is it being affected by a functional test?

What I did was the funcional test, I didn't write this test. 

I had to add this "hack" in this because when using the bundle in a normal way (by normal I mean loading `SonataPageBundle` class), it calls `Page::setSlugify` method as I said.

> Finally, since this test isn't checking all the available implementations, I think it should be renamed in order to make clear which one is intended to checked (own, "sonata.core.slugify.native", or "sonata.core.slugify.cocur").

This test I modified is in the class `tests/Model/PageTest.php`, to me it is clear that is testing the own deprecated and very unlikely to be executed implementation, IMHO that "hack" in a test that is going to be removed in the next major version is not really a big deal.

Reply:
>This test I modified is in the class tests/Model/PageTest.php, to me it is clear that is testing the own deprecated and very unlikely to be executed implementation

Since `Page::slugify()` can use any implementation based on the callable passed to `Page::setSlugifyMethod()`, I'm sorry, but it's not clear for me that the own implementation is being tested, since it depends on the value of the property you're manipulating here:

https://github.com/sonata-project/SonataPageBundle/blob/926cbbf124e214cc9b9824051b7273836635fd35/src/Model/Page.php#L200-L203

https://github.com/sonata-project/SonataPageBundle/blob/926cbbf124e214cc9b9824051b7273836635fd35/src/Model/Page.php#L590-L615

Whatever, if you think this is the way, go ahead.